include/science/synapse/device.h

```diff
@@ -66,6 +74,32 @@ class Device : public IDevice {
    */
   [[nodiscard]] auto uri() const -> const std::string&;
 
+  /**
+   * Get the logs from the device.
+   * 
+   * @param log_level Minimum log level to retrieve (default: INFO)
+   * @param since_ms Get logs from this many milliseconds ago (optional)
+   * @param start_time_ns Start time in nanoseconds since epoch (optional)
+   * @param end_time_ns End time in nanoseconds since epoch (optional)
+   * @return std::vector<std::string> containing log entries
+   */
+  [[nodiscard]] auto get_logs(
+      const std::string& log_level = "INFO",
+      std::optional<int64_t> since_ms = std::nullopt,
+      std::optional<int64_t> start_time_ns = std::nullopt,
+      std::optional<int64_t> end_time_ns = std::nullopt) -> std::vector<std::string>;
+
+  /**
+   * Tail the logs from the device.
+   * 
+   * @param log_level Minimum log level to retrieve (default: INFO)
+   * @param timeout Timeout for the request (optional)
+   * @return std::vector<std::string> containing the most recent log entries
+   */
+  [[nodiscard]] auto tail_logs(
+      const std::string& log_level = "INFO",
```

Comment:
let's add a LogLevel enum or use the existing proto one rather than accepting raw strings.

I like the default value!